Fails on darwin:

```
 D |2020-11-30 14:38:21 1606747101058826| TEST[web::server::api::ApiControllerTest]:response:
---
HTTP/1.1 200 OK
Content-Length: 5

test3
---

 I |2020-11-30 14:38:21 1606747101059128| TEST[web::server::api::ApiControllerTest]:FINISHED - success!
 I |2020-11-30 14:38:21 1606747101059414| TEST[web::server::api::ApiControllerTest]:1090(micro), 68(objs)

 I |2020-11-30 14:38:21 1606747101059724| TEST[web::server::handler::AuthorizationHandlerTest]:START...
 V |2020-11-30 14:38:21 1606747101060020| TEST[web::server::handler::AuthorizationHandlerTest]:header="Basic Zm9vOmJhcg==" -> user="foo" password="bar"
 V |2020-11-30 14:38:21 1606747101060313| TEST[web::server::handler::AuthorizationHandlerTest]:header="Basic Zm9vOmJhcg==" -> user="foo" password="bar"
 I |2020-11-30 14:38:21 1606747101060599| TEST[web::server::handler::AuthorizationHandlerTest]:FINISHED - success!
 I |2020-11-30 14:38:21 1606747101060935| TEST[web::server::handler::AuthorizationHandlerTest]:593(micro), 15(objs)

 I |2020-11-30 14:38:21 1606747101061246| TEST[web::PipelineTest]:START...
 D |2020-11-30 14:38:21 1606747101061829| ClientServerTestRunner:Running server on port 0. Timeout 600000000(micro)
 D |2020-11-30 14:38:21 1606747101062393| TEST[web::PipelineTest]:want to Receive 294000 bytes
 D |2020-11-30 14:38:21 1606747101062623| TEST[web::PipelineTest]:Sending 183000 bytes
 D |2020-11-30 14:38:21 1606747101105735| ClientServerTestRunner:Finished with time 43903(micro). Stopping server...
 I |2020-11-30 14:38:22 1606747102116478| TEST[web::PipelineTest]:FINISHED - success!
 I |2020-11-30 14:38:22 1606747102116876| TEST[web::PipelineTest]:1054920(micro), 30307(objs)

 I |2020-11-30 14:38:22 1606747102117166| TEST[web::PipelineTest]:START...
 E |2020-11-30 14:38:22 1606747102119205| [oatpp::network::tcp::server::ConnectionProvider::instantiateServer()]:Error. Call to getaddrinfo() failed with result=8: Operation not permitted
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: [oatpp::network::tcp::server::ConnectionProvider::instantiateServer()]: Error. Call to getaddrinfo() failed.


0% tests passed, 1 tests failed out of 1

Total Test time (real) =  24.15 sec

The following tests FAILED:
          1 - oatppAllTests (Child aborted)
Errors while running CTest
make: *** [Makefile:149: test] Erro
```

--- a/pkgs/development/libraries/oatpp/default.nix
+++ b/pkgs/development/libraries/oatpp/default.nix
@@ -1,0 +1,29 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  pname = "oatpp";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "oatpp";
+    repo = "oatpp";
+    rev = version;
+    sha256 = "05rm0m5zf1b5ky8prf6yni2074bz6yjjbrc2qk96fb48fc1198gw";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  # Tests fail on darwin. See https://github.com/NixOS/nixpkgs/pull/105419#issuecomment-735826894
+  doCheck = !stdenv.isDarwin;
+
+  meta = with stdenv.lib; {
+    homepage = "https://oatpp.io/";
+    description = "Light and powerful C++ web framework for highly scalable and resource-efficient web applications";
+    license = licenses.asl20;
+    maintainers = [ maintainers.ivar ];
+    platforms = platforms.all;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6110,6 +6110,8 @@ in
 
   oathToolkit = callPackage ../tools/security/oath-toolkit { };
 
+  oatpp = callPackage ../development/libraries/oatpp { };
+
   obex_data_server = callPackage ../tools/bluetooth/obex-data-server { };
 
   obexd = callPackage ../tools/bluetooth/obexd { };

